include/mapped_soa.h

```diff
@@ -85,6 +85,27 @@ class MappedSoA {
     key_to_index_.erase(key);
   }
 
+  void copy(size_t index0, size_t index1) {
+    const size_t size_val = size();
+    if (size_val <= index0 || size_val <= index1) {
+      assert(false);
+
+      return;
+    }
+
+    if (index0 == index1) {
+      return;
+    }
+
+    const KeyType key0 = index_to_key_[index0];
+    const KeyType key1 = index_to_key_[index1];
+
+    key_to_index_[key1] = key_to_index_[key0];
+    index_to_key_[index1] = index_to_key_[index0];
```

Comment:
However, I am really not sure here if the key maps should also be copied.... is this what the users of mapped_soa would be expecting? :thinking:

Reply:
The key and indices should remain the same. For example:

```
// Map of String Key type to a position.
MappedSoA<std::string, Vector3> positions;
positions["Orc1"] = Vector::Zero;
positions["Orc2"] = Vector::Zero;
positions.Copy("Orc1", "Orc2");
```
In this case, the indices and keys should remain the same. The index map simply refers to the positions of Orc1 and Orc2 in the arrays, and the keys are self explanatory :)

Reply:
Note: There isn't actually a [] operation, though the more I think about it, the more I want one :D Maybe <index>[""].... I think I'm going to implement [] operator now!

Reply:
yeah, that's what I thought!  haha nice! operator[] would be pretty handy for sure.